src/test/storage/table_test.cpp

```diff
@@ -86,4 +86,65 @@ TEST_F(StorageTableTest, GetColumnIdByName) {
 
 TEST_F(StorageTableTest, GetChunkSize) { EXPECT_EQ(t.target_chunk_size(), 2u); }
 
+TEST_F(StorageTableTest, EmplaceChunk) {
+  auto int_value_segment = std::make_shared<ValueSegment<int32_t>>();
+  int_value_segment->append(4);
+  int_value_segment->append(6);
+  int_value_segment->append(3);
+
+  auto string_value_segment = std::make_shared<ValueSegment<std::string>>();
+  string_value_segment->append("Hello,");
+  string_value_segment->append("world");
+  string_value_segment->append("!");
+
+  Chunk c;
+
+  c.add_segment(int_value_segment);
+
+  // error because wrong number of columns
+  EXPECT_THROW(t.emplace_chunk(std::move(c)), std::exception);
+
+  Chunk c2;
+  c2.add_segment(int_value_segment);
+  c2.add_segment(string_value_segment);
+  EXPECT_EQ(t.chunk_count(), 1u);
+  t.emplace_chunk(std::move(c2));
+  EXPECT_EQ(t.chunk_count(), 1u);
+
+  t.append({4, "Hello,"});
+  t.append({6, "world"});
+  t.append({3, "!"});
+  EXPECT_EQ(t.chunk_count(), 3u);
+  EXPECT_EQ(t.row_count(), 6u);
+
+  Chunk c3;
+  c3.add_segment(int_value_segment);
+  c3.add_segment(string_value_segment);
+  t.emplace_chunk(std::move(c3));
+  EXPECT_EQ(t.chunk_count(), 4u);
+  EXPECT_EQ(t.row_count(), 9u);
+}
+
+TEST_F(StorageTableTest, CompressChunk) {
+  EXPECT_EQ(t.chunk_count(), 1u);
+  t.append({4, "Hello,"});
+  t.append({6, "world"});
+  t.append({3, "!"});
+  EXPECT_EQ(t.chunk_count(), 2u);
+
+  // check whether insertion was correct
+  const auto& chunk_1 = t.get_chunk(ChunkID{0});
+  const auto& segment_1 = chunk_1.get_segment(ColumnID{0});
+  const auto value_1 = (*segment_1)[0];
+  EXPECT_EQ(type_cast<int>(value_1), 4);
+
+  t.compress_chunk(ChunkID{0});
+
```

Comment:
You should check if the new segment is a DictionarySegment.
Otherwise, one could do nothing inside compress_chunk() and this test would pass.